sensors/sensorlib/sensor_app.py

```diff
@@ -0,0 +1,177 @@
+        self.__sensor: Sensor = Sensor(SensorApplication.signalTypeConfig(self.__sensorConfig['sensorType']), self.__sensorConfig['cycle']).\
+                                    noise(NoiseConfig(self.__sensorConfig['noiseMean'], self.__sensorConfig['noiseStd'])).\
+                                    wave(WaveConfig(self.__sensorConfig['waveMagnitude'], self.__sensorConfig['wavePoint'])).\
+                                    callTo(lambda x: print(x)).\
+                                    initialize()
+        self.__sensorIsStart: bool = False
+        self.flaskApp: Flask = Flask(__name__, static_url_path="")
+        self.flaskApi: Api = Api(self.flaskApp)
+        self.serverManager: Manager = Manager(self.flaskApp)
+
+    def run(self):
+        self.flaskApi.add_resource(SensorConfigAPI, '/api/v1/sensorconfig', endpoint='sensorconfig')
+        self.serverManager.add_command('runserver', CustomServer())
+        self.serverManager.run()
+
+    @property
+    def sensor(self) -> Sensor:
+        return self.__sensor
+
+    @sensor.setter
+    def sensor(self, sensor: Sensor) -> None:
+        self.__sensor = sensor
+
+    @property
+    def sensorConfig(self) -> sensorConfig:
+        return self.__sensorConfig
+
+    @sensorConfig.setter
+    def sensorConfig(self, cfg: sensorConfig) -> None:
+        self.__sensorConfig = cfg
+
+    @property
+    def sensorIsStart(self) -> bool:
+        return self.__sensorIsStart
+
+    @sensorIsStart.setter
+    def sensorIsStart(self, isStart: bool) -> None:
+        self.__sensorIsStart = isStart
+    
+    def startSensor(self) -> None:
+        self.sensor.start()
+        self.sensorIsStart = True
+
+    def stopSensor(self) -> None:
+        self.sensor.stop()
+        self.sensorIsStart = False
+
+    def updateConfigSensor(self, config: sensorConfigType) -> None:
+        if self.sensorIsStart:
+            self.stopSensor()
+
+        self.sensorConfig = config
+        self.sensor = Sensor(SensorApplication.signalTypeConfig(self.__sensorConfig['sensorType']), self.__sensorConfig['cycle']).\
+                                    noise(NoiseConfig(self.__sensorConfig['noiseMean'], self.__sensorConfig['noiseStd'])).\
+                                    wave(WaveConfig(self.__sensorConfig['waveMagnitude'], self.__sensorConfig['wavePoint'])).\
+                                    callTo(lambda x: self.__connector.send(x)).\
+                                    initialize()
+        self.startSensor()
+
+    @classmethod
+    def app(cls) -> Any:
+        return cls(DEFAULT_SENSOR_CONFIG)
+    
+    @staticmethod
+    def signalTypeConfig(typeStr: str) -> SignalType:
+        if typeStr == 'SineWave':
+            return SignalType.SINEWAVE
+        elif typeStr == 'StaticWave':
+            return SignalType.STATICWAVE
+        elif typeStr == 'SquareWave':
+            return SignalType.SQUAREWAVE
+
+sensorApp = SensorApplication.app()
```

Comment:
Don't share "sensorApp" object to everywhere. 
1. Make this class to singleton
2. Forward it by a specific method

Reply:
+1 đồng quan điểm